hazelcast/src/main/java/com/hazelcast/internal/eviction/impl/comparator/LFUEvictionPolicyComparator.java

```diff
@@ -42,4 +42,17 @@ public int compare(EvictableEntryView e1, EvictableEntryView e2) {
     public String toString() {
         return "LFUEvictionPolicyComparator{" + super.toString() + "} ";
     }
+
+    @Override
+    public final boolean equals(Object o) {
+        if (o instanceof LFUEvictionPolicyComparator) {
```

Comment:
could be simplified as `return (o instanceof LFUEvictionPolicyComparator);`

Reply:
shouldn't the implementation be like this?
```
	@Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        return this.getClass().equals(obj.getClass());
    }
```

Otherwise subclasses can be counted as equal?

Reply:
In addition to this, shall we make `LFUEvictionPolicyComparator` final so that there is no subclasses?

Reply:
`LFUEvictionPolicyComparator` is not in public api. Also in some places, i subclassed them for testing purposes.